src/internal/packager/helm/zarf.go

```diff
@@ -124,13 +128,20 @@ func (h *Helm) UpdateZarfAgentValues(ctx context.Context) error {
 	defer spinner.Stop()
 
 	// Force pods to be recreated to get the updated secret.
-	err = h.cluster.DeletePods(
-		ctx,
-		k8s.PodLookup{
-			Namespace: cluster.ZarfNamespaceName,
-			Selector:  "app=agent-hook",
-		},
-	)
+	deleteGracePeriod := int64(0)
```

Comment:
We should explain why no grace period is given to the pod.

Reply:
I can't think of a reason, seems especially weird for the agent which could be in the middle of a request

Reply:
I have added TODOs to both the delete grace periods, lets not change behavior here and instead resolve it in a separate PR.